<?xml version="1.0" ?>

<launch>


    <arg name="robot_type"         default="humanoid"/>
    <arg name="rviz"                default="true" />
    <arg name="robot_version"      default="$(optenv ROBOT_VERSION 40)"/>

    <!-- load the robot version manager -->
    <include file="$(find humanoid_controllers)/launch/robot_version_manager.launch">
      <arg name="robot_version" value="$(arg robot_version)"/>
      <arg name="rviz" value="$(arg rviz)"/>
    </include>
    <arg name="multiplot"          default="false"/>
    <arg name="use_external_mpc"   default="true"/>
    <arg name="wbc_frequency"      default="500"/>
    <arg name="sensor_frequency"   default="500"/>
    <arg name="use_joystick"       default="true"/>
    <arg name="with_estimation"    default="true"/>
    <arg name="use_joint_filter"   default="false"/>
    <arg name="estimator_type"     default="1"/>
    <arg name="wbc_only"           default="false"/>
    <arg name="play_back"          default="true"/>  
    <arg name="joystick_type"      default="bt2"/>


    <!-- The reference related config file of the robot -->

    <!-- The file defining gait definition -->



    <!-- multiplot -->
    <group if="$(arg multiplot)">
      <include file="$(find humanoid_interface_ros)/launch/multiplot.launch"/>
    </group>

    <!-- make the files into global parameters -->
    <param name="robot_version"     value="$(arg robot_version)"/>
    <param name="multiplot"         value="$(arg multiplot)"/>
    <param name="wbc_only"          value="$(arg wbc_only)"/>
    <param name="use_external_mpc"  value="$(arg use_external_mpc)"/>
    <param name="wbc_frequency"     value="$(arg wbc_frequency)"/>
    <param name="sensor_frequency"  value="$(arg sensor_frequency)"/>
    <param name="with_estimation"   value="$(arg with_estimation)"/>
    <param name="use_joint_filter"  value="$(arg use_joint_filter)"/>
    <param name="estimator_type"    value="$(arg estimator_type)"/>
    <param name="play_back"         value="$(arg play_back)"/>
    <param name="joystick_type"  value="$(arg joystick_type)"/>

    <param name="build_cppad_state"   value="0"/>  <!-- build_cppad_state is set to 0(waiting), 1(building), 2(done) -->

    <!-- nodelet manager -->
     <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" respawn="false" output="screen" required="true" />

    <node pkg="nodelet" type="nodelet" name="nodelet_player" args="load HumanoidPlaybackNodelet nodelet_manager" respawn="false" output="screen" required="true">
      </node>
    <!-- <node pkg="mujoco_sim" type="humanoid_sim.py" name="humanoid_sim"
          output="screen"/> -->
   
    <!-- <include file="$(find humanoid_interface)/launch/rosbag_nodelet.launch"/> -->

</launch>
